### PR TITLE
DAOS-2869 DAOS shell network scan output

### PR DESCRIPTION
**Pull request by jbrosenz**
Added network scan and network list commands that
	mirror the functionality in daos_server.

Signed-off-by: Joel Rosenzweig <joel.b.rosenzweig@intel.com>

### CODE REVIEW

**Comment by jbrosenz** [comment]
This is a work in progress that does not build.  Currently working to fix build errors in the plumbing so I can make the two new gRPCs functional.

**Comment by jbrosenz** [comment]
Initial set of build errors:

# github.com/daos-stack/daos/src/control/client
build/src/control/src/github.com/daos-stack/daos/src/control/client/connect.go:464:3: cannot use connList literal (type *connList) as type Connect in return argument:
        *connList does not implement Connect (missing RequestDeviceScanStreamer method)
build/src/control/src/github.com/daos-stack/daos/src/control/client/mocks.go:314:9: cannot use mockMgmtSvcClient literal (type *mockMgmtSvcClient) as type mgmt.MgmtSvcClient in return argument:
        *mockMgmtSvcClient does not implement mgmt.MgmtSvcClient (missing SmdListPools method)
build/src/control/src/github.com/daos-stack/daos/src/control/client/mocks.go:406:3: cannot use connList literal (type *connList) as type Connect in return argument:
        *connList does not implement Connect (missing RequestDeviceScanStreamer method)
scons: *** Error 2
scons: *** [build/src/control/bin/dmg] Error 2
# github.com/daos-stack/daos/src/control/server
build/src/control/src/github.com/daos-stack/daos/src/control/server/ctl_storage_rpc.go:173:3: ambiguous selector c.log
build/src/control/src/github.com/daos-stack/daos/src/control/server/ctl_storage_rpc.go:218:3: ambiguous selector c.log
build/src/control/src/github.com/daos-stack/daos/src/control/server/ctl_storage_rpc.go:226:5: ambiguous selector c.log
build/src/control/src/github.com/daos-stack/daos/src/control/server/ctl_storage_rpc.go:249:3: ambiguous selector c.log
build/src/control/src/github.com/daos-stack/daos/src/control/server/ctl_storage_rpc.go:281:3: ambiguous selector c.log
build/src/control/src/github.com/daos-stack/daos/src/control/server/ctl_storage_rpc.go:317:3: ambiguous selector c.log
scons: *** Error 2

**Comment by jbrosenz** [comment]
Current build errors:

`build/src/control/src/github.com/daos-stack/daos/src/control/client/connect.go:464:3: cannot use connList literal (type *connList) as type Connect in return argument:
        *connList does not implement Connect (missing RequestDeviceScanStreamer method)
build/src/control/src/github.com/daos-stack/daos/src/control/client/mocks.go:423:3: cannot use connList literal (type *connList) as type Connect in return argument:
        *connList does not implement Connect (missing RequestDeviceScanStreamer method)
# github.com/daos-stack/daos/src/control/client
build/src/control/src/github.com/daos-stack/daos/src/control/client/connect.go:464:3: cannot use connList literal (type *connList) as type Connect in return argument:
        *connList does not implement Connect (missing RequestDeviceScanStreamer method)
build/src/control/src/github.com/daos-stack/daos/src/control/client/mocks.go:423:3: cannot use connList literal (type *connList) as type Connect in return argument:
        *connList does not implement Connect (missing RequestDeviceScanStreamer method)
scons: *** Error 2
scons: *** Error 2
scons: *** [build/src/control/bin/dmg] Error 2
scons: *** [build/src/control/bin/agent] Error 2
# github.com/daos-stack/daos/src/control/server
build/src/control/src/github.com/daos-stack/daos/src/control/server/ctl_storage_rpc.go:173:3: ambiguous selector c.log
build/src/control/src/github.com/daos-stack/daos/src/control/server/ctl_storage_rpc.go:218:3: ambiguous selector c.log
build/src/control/src/github.com/daos-stack/daos/src/control/server/ctl_storage_rpc.go:226:5: ambiguous selector c.log
build/src/control/src/github.com/daos-stack/daos/src/control/server/ctl_storage_rpc.go:249:3: ambiguous selector c.log
build/src/control/src/github.com/daos-stack/daos/src/control/server/ctl_storage_rpc.go:281:3: ambiguous selector c.log
build/src/control/src/github.com/daos-stack/daos/src/control/server/ctl_storage_rpc.go:317:3: ambiguous selector c.log
scons: *** Error 2
scons: *** [build/src/control/bin/daos_server] Error 2
`

File: src/control/client/connect.go [hunk 1]
```diff
@@ -302,6 +302,9 @@ type Connect interface {
 	PoolDestroy(*PoolDestroyReq) error
 	BioHealthQuery(*mgmtpb.BioHealthReq) ResultQueryMap
 	SmdListDevs(*mgmtpb.SmdDevReq) ResultSmdMap
+// this is just a guess.  I'm not sure how these two functions should be defined
+	RequestProviderList(*ctlpb.ProviderListRequest) (*ctlpb.ProviderListReply, error)
+	RequestDeviceScanStreamer(*ctlpb.DeviceScanRequest, *ctlpb.MgmtCtl_RequestDeviceScanStreamerServer) error
```
Comment (tanabarr) [inline src/control/client/connect.go]: the streaming should be encapsulated within the client method ideally and this streaming server not exported

**Comment by jbrosenz** [comment]
Interim build, not ready for review.

**Comment by jbrosenz** [comment]
No longer needed.

**Pull request closed without merging**